common/conf/platform/boost_filesystem/paths.cpp

```diff
@@ -28,24 +28,24 @@ using namespace std;
 namespace conf = mender::common::conf;
 namespace fs = boost::filesystem;
 
-const string DefaultPathConfDir =
-	conf::GetEnv("MENDER_CONF_DIR", (fs::path("/etc") / "mender").string());
-const string DefaultPathDataDir =
-	conf::GetEnv("MENDER_DATA_DIR", (fs::path("/usr/share") / "mender").string());
-const string DefaultDataStore =
-	conf::GetEnv("MENDER_DATASTORE_DIR", (fs::path("/var/lib") / "mender").string());
+string Join(const string &prefix, const string &suffix) {
+	return (fs::path(prefix) / suffix).string();
+}
+
+const string DefaultPathConfDir = conf::GetEnv("MENDER_CONF_DIR", Join("/etc", "mender"));
+const string DefaultPathDataDir = conf::GetEnv("MENDER_DATA_DIR", Join("/usr/share", "mender"));
+const string DefaultDataStore = conf::GetEnv("MENDER_DATASTORE_DIR", Join("/var/lib", "mender"));
 const string DefaultKeyFile = "mender-agent.pem";
 
-const string DefaultConfFile = (fs::path(DefaultPathConfDir) / "mender.conf").string();
-const string DefaultFallbackConfFile = (fs::path(DefaultDataStore) / "mender.conf").string();
+const string DefaultConfFile = Join(DefaultPathConfDir, "mender.conf");
+const string DefaultFallbackConfFile = Join(DefaultDataStore, "mender.conf");
 
 // device specific paths
-const string DefaultArtScriptsPath = (fs::path(DefaultDataStore) / "scripts").string();
-const string DefaultRootfsScriptsPath = (fs::path(DefaultPathConfDir) / "scripts").string();
-const string DefaultModulesPath = (fs::path(DefaultPathDataDir) / "modules" / "v3").string();
-const string DefaultModulesWorkPath = (fs::path(DefaultDataStore) / "modules" / "v3").string();
-const string DefaultBootstrapArtifactFile =
-	(fs::path(DefaultDataStore) / "bootstrap.mender").string();
+const string DefaultArtScriptsPath = Join(DefaultDataStore, "scripts");
+const string DefaultRootfsScriptsPath = Join(DefaultPathConfDir, "scripts");
+const string DefaultModulesPath = Join(DefaultPathDataDir, "modules/v3");
+const string DefaultModulesWorkPath = Join(DefaultDataStore, "modules/v3");
```

Comment:
Should `modules/v3` also be joined?

Reply:
I don't exactly know, TBH. `boost::fs::path` docs mention the [*portable format*](https://www.boost.org/doc/libs/1_39_0/libs/filesystem/doc/reference.html#Pathname-formats) for paths (basically POSIX paths) which should be accepted and properly handled. So my understanding is that by converting `DefaultDataStore` to `boost::fs::path`, appending to it should work fine with paths in the portable format. But like I said, not 100% sure here, I'd have to check it on a system that has a different path format, but no such is easily available for me.